core/templates/self_list.h

```diff
@@ -77,7 +77,8 @@ class SelfList {
 		}
 
 		void remove(SelfList<T> *p_elem) {
-			ERR_FAIL_COND(p_elem->_root != this);
+			ERR_FAIL_COND_STDOUT(p_elem->root != this);
```

Comment:
```suggestion
			ERR_FAIL_COND_STDOUT(p_elem->_root != this);
```

Reply:
I tested that with _root already, it didn't compile so changed and same result with the marco

Reply:
That shouldn't happen, if the code works with one and not the other something else is wrong

Reply:
You *have* to have a return statement after it in your example, that is required

Reply:
It must be:
```
if (p_elem->_root != this) {
     fprintf(stderr, "ERROR: Cannot remove element not in list!\n   at: %s (%s:%i)\n", FUNCTION_STR, __FILE__, __LINE__);
     return;
}
```

Reply:
Better:
```
if (unlikely(p_elem->_root != this)) {
     fprintf(stderr, "ERROR: Cannot remove element not belonging to this list!\n   at: %s (%s:%i)\n", FUNCTION_STR, __FILE__, __LINE__);
     return;
}
```

Reply:
```suggestion
			if (unlikely(p_elem->_root != this)) {
				fprintf(stderr, "ERROR: Cannot remove element not belonging to this list!\n   at: %s (%s:%i)\n", FUNCTION_STR, __FILE__, __LINE__);
				return;
			}
```
As per the above suggestion

Reply:
> I'd still like to see your errors with this check unchanged, to see what is going on there, I don't see how this can be part of the crash as, it shouldn't trigger infinite recursion as it is not part of the `MessageQueue` code, so pinning down what's wrong here is important rather than just patching it

The logs it shows is as follows
```
ERROR: Condition "p_elem->_root != this" is true.
   at: remove (./core/templates/self_list.h:80)
ERROR: Condition "p_elem->_root != this" is true.
   at: remove (./core/templates/self_list.h:80)
ERROR: Condition "p_elem->_root != this" is true.
   at: remove (./core/templates/self_list.h:80)
ERROR: Condition "p_elem->_root != this" is true.
   at: remove (./core/templates/self_list.h:80)
ERROR: Condition "p_elem->_root != this" is true.
   at: remove (./core/templates/self_list.h:80)
[OS terminates the binary after detecting an infinite loop]
```

Reply:
This is with the original code or with your change?

Reply:
With the return statement.

Reply:
Will take a look later today

Reply:
Looks like an infinite loop and not a recursion, try:
```suggestion
			DEV_ASSERT(p_elem->_root == this);
```
It should be `==`, and give me the stack trace

Reply:
My suspicion is that the error might be due to copying of `SelfList::List`, which is done in `TileMap`, will investigate more later, and how to possibly work around it